manifests/init.pp

```diff
@@ -0,0 +1,130 @@
+# @summary Manage a MySQL/MariaDB server configuration.
+#
+# @param create_mysql_home
+# Should the mysql_home be created?
+#
+# @param dbs
+# Raw data to define mysql::db resources.
+#
+# @param mysql_gid
+# The GID for the mysql user.
+#
+# @param mysql_groupname
+# The name of the mysql group.
+#
+# @param mysql_home
+# The home dir of the mysql user.
+#
+# @param mysql_logdir
+# Location of MySQL log files.
+#
+# @param mysql_parentdir
+# Location of directory that contains the mysql_home.
+# NOTE: Might be useful for this to be an array so multiple nested
+# dirs can be created as dependencies.
```

Comment:
If this is just the parent directory of `mysql_home` can't that programmatically figured out from `mysql_home`? See logic in `nfs_client`, `gpfs_client`, or `lustre::client` (and their bind mount resource types) for examples of ensuring all parent directories for a given path.

What is the purpose of setting this variable uniquely?

Reply:
That was what I was after, thanks for the tip. Fixed.